### DISCUSSION TITLE
Embedding Net Skip Dense Layer Architecture Doubt

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#2183] [Q&A] [answered]
Hi,

![dpmd_model_plot](https://user-images.githubusercontent.com/10074710/207875046-5016e401-972a-4c1d-972e-195bbf7ff899.png)

In the (f) part Why do we have (x,x) as a residual connection for embedding net but not in (g) for output energy creation?

**Comment by AnuragKr** [comment]
@njzjz Could you please help me clear the above doubt?

**Reply by njzjz** [reply]
I don't know. You could ask @wanghan-iapcm.

**Comment by AnuragKr** [comment, marked as answer]
No issue, I think I figured it out it is for making dimensions equal to output for addition. As the output is going to be a matrix for (f) and in (g) part only single dimension output so we can directly add input value of the single dimension.

Once again Thanks for attending to my question.
@wanghan-iapcm my explanation is correct?

**Reply by wanghan-iapcm** [reply]
Yes you are right, the purpose is the make the dimension of the tensors equal so they can be added.
